datadog_checks_base/tests/base/checks/test_agent_check.py

```diff
@@ -50,6 +51,46 @@ def test_load_config():
         AgentCheck.load_config(':')
 
 
+def test_parse_special_values_in_load_config():
+    assert AgentCheck.load_config("boolean: true") == {"boolean": True}
+    assert AgentCheck.load_config("boolean: false") == {"boolean": False}
+
+    assert AgentCheck.load_config("number: .inf") == {"number": float("inf")}
+    assert AgentCheck.load_config("number: .INF") == {"number": float("inf")}
+
+    assert AgentCheck.load_config("number: -.inf") == {"number": float("-inf")}
+    assert AgentCheck.load_config("number: -.INF") == {"number": float("-inf")}
+
+    assert AgentCheck.load_config("number: 0xF") == {"number": 15.0}  # Hexadecimal
+    assert AgentCheck.load_config("number: 0b1111") == {"number": 15.0}  # Binary
+
+    assert AgentCheck.load_config("number: !!int 1 ") == {"number": 1}
+    assert isinstance(AgentCheck.load_config("number: !!int 1 ")["number"], int)
+
+    assert AgentCheck.load_config("number: !!float 1 ") == {"number": 1}
+    assert isinstance(AgentCheck.load_config("number: !!float 1 ")["number"], float)
+
+    # Check that "inf" is still a string
+    assert AgentCheck.load_config("string: inf") == {"string": "inf"}
+    assert isinstance(AgentCheck.load_config("string: inf")["string"], str)
+
+    # Check that quoted ".inf" is still a string
+    assert AgentCheck.load_config("string: \".inf\"") == {"string": ".inf"}
+    assert isinstance(AgentCheck.load_config("string: \".inf\"")["string"], str)
+
+    # nan values are not equal to themselves
+    config = AgentCheck.load_config("number: .nan")
+    assert "number" in config
+    assert math.isnan(config["number"])
+
+    # Assert that a string with an inf inside it is parsed as a string
+    assert AgentCheck.load_config("string: \"hi inf\"") == {"string": "hi inf"}
+    assert AgentCheck.load_config("string: hi inf") == {"string": "hi inf"}
+    assert AgentCheck.load_config("string: \"this inf is in the middle\"") == {"string": "this inf is in the middle"}
+    assert AgentCheck.load_config("string: this inf is in the middle") == {"string": "this inf is in the middle"}
+    assert AgentCheck.load_config("string: infinity") == {"string": "infinity"}
```

Comment:
Maybe we can move these tests with actual python dictionaries to a test module specifically to test the new module that contains this logic? Just thinking we might want to have that logic tested independently of the rest of the method that reads the yaml, which for each test would launch a subprocess I think, I haven't looked to see if there are any mocks for this subprocess call.